.buildkite/scripts/common.sh

```diff
@@ -69,3 +71,32 @@ with_Terraform() {
     chmod +x ${WORKSPACE}/terraform
     terraform version
 }
+
+google_cloud_auth() {
+    secretFileLocation=$(mktemp -d -p "${WORKSPACE}" -t "${TMP_FOLDER_TEMPLATE_BASE}.XXXXXXXXX")/google-cloud-credentials.json
+    echo "${PRIVATE_CI_GCS_CREDENTIALS_SECRET}" > ${secretFileLocation}
+    gcloud auth activate-service-account --key-file ${secretFileLocation} 2> /dev/null
+    export GOOGLE_APPLICATIONS_CREDENTIALS=${secretFileLocation}
+}
+
+upload_packages_to_gcp_bucket() {
+    pattern=${1}
+    baseUri="gs://${JOB_GCS_BUCKET}/${REPO}"
+    bucketUriCommit="${baseUri}"/commits/${BUILDKITE_COMMIT}
+    bucketUriDefault="${baseUri}"/snapshots
```

Comment:
This probably needs to be prefixed by buildkite too ? @sharbuz 

```suggestion
    bucketUriDefault="${baseUri}"/buildkite-snapshots
```

Reply:
I'm not sure where this PATH is used in other systems/services, perhaps it will need to change too.
We can easily change this line:
`bucketUriCommit="${baseUri}"/commits/${BUILDKITE_COMMIT}`
to
`bucketUriCommit="${baseUri}"/commits/buildkite-${BUILDKITE_COMMIT}`
WDYT @mrodm ?

Reply:
Probably it's better to prefix both commits and snapshots with `buildkite-` until the migration to Buildkite is completed ?

Once it is done and Jenkins does not run, all these paths should be the same as they were in Jenkins (without the prefix). Same for line https://github.com/elastic/fleet-server/pull/2628/files#diff-68c0021e761ebe6b8a42d32dfdf3dda2200f0f2fee50106c85c9aea530ccf3c3R89

Reply:
Probably in this case is better to use the next approach:
```
bucketUriCommit="${baseUri}"/**buildkite**/commits/${BUILDKITE_COMMIT}
bucketUriDefault="${baseUri}"/**buildkite**/snapshots
bucketUriDefault="${baseUri}"/**buildkite**/pull-requests/pr-${GITHUB_PR_NUMBER}
```
WDYT @mrodm ?

Reply:
:+1: 

Even you could add this directly to the `baseUri` variable, so it is just in one place for all the other variables.

```
baseUri="gs://${JOB_GCS_BUCKET}/${REPO}/buildkite"
```

Please, add a TODO comment to remove it once it is completed the migration.
It looks like there would be some tasks to do for the final migration.

Reply:
done, please check it

Reply:
thanks!